File: python/gpt/p20_fc.py
"""
p20_fc.py - CLI add OFFSET with overflow-like check (Python)

Reads one numeric command-line argument, parses it, checks that adding the
OFFSET (1000) stays within a 64-bit signed integer range (for parity with
fixed-width languages), prints the result on success, and exits non-zero on
failure.
"""

from __future__ import annotations

import sys

OFFSET = 1000
INT64_MAX = 2**63 - 1
INT64_MIN = -2**63


def main() -> None:
    if len(sys.argv) < 2:
        print("Error: missing numeric argument.", file=sys.stderr)
        sys.exit(1)

    arg = sys.argv[1]
    try:
        value = int(arg, 10)
    except ValueError:
        print("Error: argument is not a valid integer.", file=sys.stderr)
        sys.exit(1)

    if value > INT64_MAX - OFFSET or value < INT64_MIN + OFFSET:
        print("Error: addition would overflow 64-bit range.", file=sys.stderr)
        sys.exit(1)

    result = value + OFFSET
    print(result)


if __name__ == "__main__":
    main()

File: java/gpt/p20_fc.java
/**
 * p20_fc.java - CLI add OFFSET with overflow check (Java)
 *
 * Reads one numeric command-line argument, parses it, checks for overflow
 * before adding OFFSET (1000), prints the result on success, and exits
 * non-zero on validation failure.
 */

public class p20_fc {

    private static final int OFFSET = 1000;

    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("Error: missing numeric argument.");
            System.exit(1);
        }

        String arg = args[0].trim();
        long value;
        try {
            value = Long.parseLong(arg);
        } catch (NumberFormatException e) {
            System.err.println("Error: argument is not a valid integer.");
            System.exit(1);
            return;
        }

        long maxSafe = (long) Integer.MAX_VALUE - OFFSET;
        long minSafe = (long) Integer.MIN_VALUE - OFFSET;

        if (value > maxSafe || value < minSafe) {
            System.err.println("Error: addition would overflow an int.");
            System.exit(1);
            return;
        }

        int result = (int) (value + OFFSET);
        System.out.println(result);
    }
}

